File: pine-java/src/main/java/page/liam/pine/operators/TransformSize.java
package page.liam.pine.operators;

import page.liam.pine.AbstractOperator;
import page.liam.pine.CancellationToken;
import page.liam.pine.OperatorParams;
import page.liam.pine.OperatorInput;
import page.liam.pine.OperatorOutput;

import java.util.Map;

/**
 * Operator: transform_size
 * Metadata contract
 *   CommonInput:  []
 *   CommonOutput: [<target_field>]
 *   ItemInput:    []
 *   ItemOutput:   []
 */
public class TransformSize extends AbstractOperator implements page.liam.pine.ConcurrentSafe, page.liam.pine.ConsumesRowSet {
    @Override
    public void init(OperatorParams params) {}

    @Override
    public void execute(CancellationToken token, OperatorInput input, OperatorOutput output) {
        output.setCommon(commonOutput().get(0), input.itemCount());
    }
}

File: pine-python/pine/operators/transform_size.py
from __future__ import annotations

from pine.cancellation import CancellationToken
from pine.operator import (
    AbstractOperator,
    ConcurrentSafe,
    ConsumesRowSet,
    OperatorInput,
    OperatorOutput,
    OperatorParams,
)


class TransformSize(AbstractOperator, ConcurrentSafe, ConsumesRowSet):
    def __init__(self):
        pass

    def init(self, params: OperatorParams):
        pass

    def execute(
        self, token: CancellationToken, input_: OperatorInput,
        output: OperatorOutput,
    ) -> None:
        output.set_common(self.common_output()[0], input_.item_count())
